StudentAppMvc/StudentAppMvc/Controllers/ThtrStudentController.cs

```diff
@@ -0,0 +1,189 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using StudentAppMvc.Exceptions;
+using StudentAppMvc.Filter;
+using StudentAppMvc.Models;
+using StudentAppMvc.Models.ViewModels;
+using StudentAppMvc.Services;
+
+namespace StudentAppMvc.Controllers
+{
+    //[AuthenticationFilter]
+    [LogginFilter]
+
+    public class ThtrStudentController : Controller
+    {
+        private static StudentListViewModel _studentListViewModel;
+
+        private ISubjectService _subjectService;
+        private IStudentService _studentService;
+        private ValidationService _validationService;
+        public ThtrStudentController(IStudentService studentService, ISubjectService subjectService, ValidationService validationService)
+        {
+            _subjectService = subjectService;
+            _studentService = studentService;
+            _validationService = validationService;
+        }
+
+        // GET: StudentController
+        public ActionResult Index()
+        {
+            List<StudentMarkDTO> studentMarkList = _studentService.GetStudentMarkList();
+            //if (lastestCount > 0 && studentMarkList != null && studentMarkList.Count >= 5)
+            //{
+            //    studentMarkList = studentMarkList.OrderByDescending(st => st.StudentId).ToList().GetRange(0, 5);
+            //}
+
+            return View(new StudentListViewModel(studentMarkList));
+        }
+
+
+
+        /// <summary>
+        /// Search StudentMarkViewModel by searchName and Gender
+        /// </summary>
+        /// <param name="searchName"></param>
+        /// <param name="searchGender"></param>
+        /// <returns></returns>
+        public ActionResult Search(string searchName = "", string searchGender = "")
+        {
+            StudentListViewModel studentListDTO = new StudentListViewModel(_studentService.Search(searchName, searchGender), searchName: searchName, searchGender: searchGender);
+            return View(studentListDTO);
+        }
+
+
+        // GET: StudentController/Details/5
+        public ActionResult Details(int id)
+        {
+            Student student = _studentService.GetById(id);
+            ViewBag.Student = student;
```

Comment:
remove this line. Good practice is not to use ViewBag, except for special case as it's not transparent enough. Of course, we need to pracice ViewBag to understand it, but it should be in other cases if possible.